### PR TITLE
Address comments from PR

### PR DESCRIPTION
**Pull request by proteneer**
@maxentile

### CODE REVIEW

**Review by maxentile — APPROVED** [review]
Nice! Thanks.

Addresses https://github.com/proteneer/timemachine/pull/890#discussion_r1005919999 , https://github.com/proteneer/timemachine/pull/890#discussion_r1005929912 , https://github.com/proteneer/timemachine/pull/890#discussion_r1005931748

**Comment by proteneer** [comment]
I will fix the `or` statement - sorry I just brainfarted on it

**Comment by proteneer** [comment]
additional changes:
1) use `and` as opposed to `or` in lambda check
2) refactor `construct_pre_optimized_relative_lambda_schedule` (which will eventually be replaced with a `np.linspace`)
3) disable all edges from `hif2a` set, and replace with cherry picked ones that are bad (set to nightly)

File: timemachine/fe/rbfe.py [hunk 1]
```diff
@@ -617,12 +636,7 @@ def estimate_relative_free_energy(
     single_topology = SingleTopology(mol_a, mol_b, core, ff)
 
     if lambda_schedule is None:
-        lambda_schedule = np.array(
-            [0.0, 0.02, 0.04, 0.06, 0.07, 0.08, 0.11, 0.13, 0.15, 0.17, 0.18, 0.20, 0.25, 0.32, 0.42]
-        )
-        lambda_schedule = np.concatenate([lambda_schedule, (1 - lambda_schedule[::-1])])
-        if n_windows:
-            lambda_schedule = interpolate_pre_optimized_protocol(lambda_schedule, n_windows)
+        lambda_schedule = construct_pre_optimized_relative_lambda_schedule(n_windows)
```
Comment (badisa) [inline timemachine/fe/rbfe.py:639]: Isn't the default here that n_windows is None? Don't see anything that protects against this.

Comment (proteneer) [reply timemachine/fe/rbfe.py:639]: good catch

Comment (proteneer) [reply timemachine/fe/rbfe.py:639]: fixed in [866b5fb](https://github.com/proteneer/timemachine/pull/894/commits/866b5fbc0f1316fef016ece171a3a797afef1a70)

File: timemachine/fe/rbfe.py [hunk 2]
```diff
@@ -540,10 +555,14 @@ def optimize_coordinates(initial_states):
...
 
+    all_coords = []
+
     for state, coords in zip(initial_states, all_xs):
         # sanity check that no atom has moved more than 7 angstroms away (arbitrary)
         assert np.amax(np.linalg.norm(state.x0 - coords, axis=1)) < 0.7
-        state.x0 = coords
+        all_coords.append(coords)
+
+    return all_coords
```
Comment (maxentile) [inline timemachine/fe/rbfe.py:565]: Nit: why not just `return all_xs`?

File: tests/test_single_topology_confgen.py [hunk 3]
```diff
@@ -80,21 +79,40 @@ def run_edge(mol_a, mol_b, protein_path):
...
-            run_edge(mol_a, mol_b, protein_path)
+    n_windows = 16
+
+    for src, dst in edges:
+        print("\nProcessing", src, "->", dst, "\n")
+        mol_a = get_mol_by_name(mols, src)
+        mol_b = get_mol_by_name(mols, dst)
+        # try both directions
+        run_edge(mol_a, mol_b, protein_path, n_windows)
+        run_edge(mol_b, mol_a, protein_path, n_windows)
```
Comment (maxentile) [inline tests/test_single_topology_confgen.py:118]: Nit, testing -- still may be useful to add a unit test of `optimize_coordinates`, but this reduction of the nightly integration tests from 81 directed edges to 18*2=36 directed edges is good, and the annotations of why the selected edges were difficult is very helpful! 👍

File: timemachine/fe/lambda_schedule.py [hunk 4]
```diff
@@ -83,3 +83,28 @@ def construct_pre_optimized_absolute_lambda_schedule_solvent(num_windows, nonbon
...
+def construct_pre_optimized_relative_lambda_schedule(n_windows):
+    """
+    Interpolate a pre-optimized schedule for relative binding free energies, typically used for both complex
+    and solvent. Generated by manually staring at regions with poor overlap on hif2a runs. The longer term
+    goal is to convert this into an evenly spaced schedule (eg. np.linspace(0, 1, n_windows)) and have the
+    parameter interpolation code do non-linear setup of the the parameters.
+
+    Parameters
+    ----------
+    n_windows: int
```
Comment (maxentile) [inline timemachine/fe/lambda_schedule.py:97]: optional int, if None will return 30 windows

**Review by maxentile — APPROVED** [review]

File: timemachine/fe/rbfe.py (hunk 2 above)
Comment (proteneer) [reply timemachine/fe/rbfe.py:565]: lack of sleep, will fix

File: timemachine/fe/lambda_schedule.py (hunk 4 above)
Comment (proteneer) [reply timemachine/fe/lambda_schedule.py]: willfix

File: tests/test_single_topology_confgen.py (hunk 3 above)
Comment (proteneer) [reply tests/test_single_topology_confgen.py:118]: Also added a single unit test here as well.

File: timemachine/md/minimizer.py [hunk 5]
```diff
@@ -374,25 +385,30 @@ def val_and_grad_fn_bfgs(x_local_flattened):
...
-        per_atom_force_norms = np.linalg.norm(forces, axis=1)
         argmax_local = np.argmax(per_atom_force_norms)
         worst_atom_idx = local_idxs[argmax_local]
         print(f"atom with highest force norm after minimization: {worst_atom_idx}")
         print(f"force(x_final)[{worst_atom_idx}] = {forces[argmax_local]}")
-
         print("-" * 70)
 
+    # note that this over the local atoms only, as this function is not concerned
+    assert np.amax(per_atom_force_norms) < MAX_FORCE_NORM
```
Comment (maxentile) [inline timemachine/md/minimizer.py:402]: Nice -- this seems like a confident assertion, and better aligns with the intent of `local_minimize`.

(Don't necessarily want an assertion error if the user happens to pass in a minimized structure / if line search can't find a step that satisfies "sufficient decrease" conditions, but do always want an assertion error if the result is unsuitable to use as an initial condition for MD.)

**Review by maxentile — APPROVED** [review]

**Pull request merged**